diff --git a/bin/.java_wrapper b/bin/.java_wrapper
index 0ccbdfb..51a53a4 100755
--- a/bin/.java_wrapper
+++ b/bin/.java_wrapper
@@ -11,7 +11,7 @@ PATH=/bin:/usr/bin
 
 progname=`basename $0`
 case "`uname -m`" in
-    i[3-6]86  | ia32 | ia64 | i?86)
+    i[3-6]86 | ia32 | ia64 | i?86 | x86_64)
         proc=i386
         ;;
     sparc*)
diff --git a/jre/bin/.java_wrapper b/jre/bin/.java_wrapper
index 0ccbdfb..51a53a4 100755
--- a/jre/bin/.java_wrapper
+++ b/jre/bin/.java_wrapper
@@ -11,7 +11,7 @@ PATH=/bin:/usr/bin
 
 progname=`basename $0`
 case "`uname -m`" in
-    i[3-6]86  | ia32 | ia64 | i?86)
+    i[3-6]86 | ia32 | ia64 | i?86 | x86_64)
         proc=i386
         ;;
     sparc*)
diff --git a/jre/bin/realpath b/jre/bin/realpath
index 6b7c1a5..65505b8 100755
--- a/jre/bin/realpath
+++ b/jre/bin/realpath
@@ -7,7 +7,7 @@
 SVPATH="$PATH"
 PATH=/bin:/usr/bin
 case "`uname -m`" in
-    i[3-6]86)
+    i[3-6]86 | ia32 | ia64 | i?86 | x86_64)
         proc=i386
         ;;
     sparc*)
